core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -151,12 +153,15 @@ public InnerHitBuilder innerHit() {
     }
 
     public HasChildQueryBuilder innerHit(InnerHitBuilder innerHit) {
-        innerHit.setParentChildType(type);
-        innerHit.setQuery(query);
+        innerHit.setParentChildType(type, query);
```

Comment:
It seem to me that it would be possible to not call `setParentChildType` here, but to make a copy in `extractInnerHitBuilders` and call `setParentChildType` there on the copy? This way we would never modify a setter argument?